lib/elixir/lib/kernel.ex

```diff
@@ -3558,16 +3558,33 @@ defmodule Kernel do
 
   """
   defmacro use(module, opts \\ []) do
-    expanded = Macro.expand(module, __CALLER__)
-
-    case is_atom(expanded) do
-      false ->
-        raise ArgumentError, "invalid arguments for use, expected an atom or alias as argument"
-      true ->
+    fun = fn
+      expanded when is_atom(expanded) ->
         quote do
           require unquote(expanded)
           unquote(expanded).__using__(unquote(opts))
         end
+      _otherwise ->
+        raise ArgumentError, "invalid arguments for use, expected an atom or alias as argument"
+    end
+
+    case module do
```

Comment:
Try moving the case to a separate function that returns a list. Then you can just do: `Enum.map parse_aliases(module, __CALLER__), fun` with unquote_splicing at the end.